app/services/forest_liana/authentication.rb

```diff
@@ -0,0 +1,66 @@
+module ForestLiana
+  class Authentication
+    def initialize()
+      @oidc_client_manager_service = ForestLiana::OidcClientManager.new()
+      @authorization_getter = ForestLiana::AuthorizationGetter.new()
+      @token_service = ForestLiana::Token.new()
+    end
+
+    def _parse_state(state)
```

Comment:
[MISC] In ruby private functions are not preceded by underscores. Instead in a single file you separate private functions from the other with the `private` keyword: https://jakeyesbeck.com/2016/01/24/ruby-private-class-methods/